app/src/main/java/eu/kanade/tachiyomi/ui/reader/viewer/pager/PagerViewer.kt

```diff
@@ -210,6 +217,26 @@ abstract class PagerViewer(val activity: ReaderActivity) : BaseViewer {
         }
     }
 
+    /**
+     * Switch to DownloadPageLoader when the chapter is downloaded
+     */
+    private fun verifyIfShouldSwitchToDownloadLoader(currentReaderChapter: ReaderChapter) {
+        val loader = currentReaderChapter.pageLoader ?: return
+        if (loader is HttpPageLoader) {
+            val downloadManager = Injekt.get<DownloadManager>()
```

Comment:
make the downloadManager in readerpresenter non-private and use it here, instead of constantly fetching it